### ISSUE TITLE
Unexpected drift on mist_device_gateway resource after manual changes via UI

### ISSUE THREAD

**Issue by zbuchheit** [Juniper/terraform-provider-mist#54]
When I make a change to a mist_device_gateway resource via the UI, I am seeing some unexpected drift. Particularly on `oob_ip_config`. 

This is my example program
```tf
resource "mist_device_gateway" "gateway_one" {
  name      = "gateway_one"
  device_id = "redacted"
  site_id   = "redacted"
  oob_ip_config = {
    type = "dhcp"
  }
  additional_config_cmds = ["#", "###"]
  dns_servers            = ["8.8.8.8"]
  managed                = false
  dhcpd_config = {
    # enabled = true
    config = {
      "test" = {
        dns_servers = ["8.8.8.8", "1.1.1.1"]
        gateway = "{{branch_ip}}.1"
        ip_end = "192.0.2.255"
        ip_start = "192.0.2.2"
        type = "local"
      }
    }
  }
}
```

If I go into the UI, and switch dhcpd_config from enabled to disabled, hit save, and change it back to enabled, hit save, I will have drift and `terraform apply` will show wanting to enact the following change.

```
  # mist_device_gateway.gateway_one will be updated in-place
  ~ resource "mist_device_gateway" "gateway_one" {
        name                   = "gateway_one"
      ~ oob_ip_config          = {
          - node1                     = {
              - type                      = "dhcp" -> null
              - use_mgmt_vrf              = false -> null
              - use_mgmt_vrf_for_host_out = false -> null
            } -> null
            # (3 unchanged attributes hidden)
        }
        # (14 unchanged attributes hidden)
    }

```

**Comment by tmunzer** [comment]
Hi @zbuchheit ,

Yes, I saw that too. I may not be able to exactly mimic the UI. Saving a configuration change from the UI can generate/remove fields with default values in a different way the provider is behaving.

I can fix this specific one by removing the default values, but there is a high possibility this will occur with other parts of the configuration.

**Comment by zbuchheit** [comment]
yeah, with the complexity of the object, I assume that eliminating all drift via manual changes is not an easy pursuit and the likely solution is to minimize the amount of manual manipulation via UI. 

I witnessed some similar behavior on the WLAN. I think I have some workarounds or pathways there.

**Comment by tmunzer** [comment]
Hi @zbuchheit,

this should have been fixed in v0.2.7. Can you test and validate?

Thanks

**Comment by zbuchheit** [comment]
this does appear to be fixed in the latest version

**Comment by tmunzer** [comment]
Thanks @zbuchheit 

> yeah, with the complexity of the object, I assume that eliminating all drift via manual changes is not an easy pursuit and the likely solution is to minimize the amount of manual manipulation via UI.
> 
> I witnessed some similar behavior on the WLAN. I think I have some workarounds or pathways there.

I confirm, I saw this for several resources too.. This is on my todo list to improve the default values to mimic the UI behavior.

**Comment by zbuchheit** [comment]
@tmunzer I have a branch for terraform-provider-mist I have been testing adding api logging to which I think should help to triage stuff further. I opened an issue in https://github.com/apimatic/go-core-runtime/issues/93 since that appears to be what you are using. When I tested the response body wasn't being logged correctly no matter how I configured the logger. Once I have something working I'll open a PR. 😄
